### ISSUE TITLE
Drag and drop from nemo-desktop to nemo window doesn't work (Ubuntu 24.04, Gnome 46, nautilus removed)

### ISSUE THREAD

**Issue by gregoiregentil** [linuxmint/nemo#3415]
### Distribution

Ubuntu 24.04

### Package version

6.0.2

### Frequency

Always

### Bug description

As the title said, drag and drop a file from the desktop (aka. nemo-desktop) to a nemo window doesn't work. The file stays on the desktop. I have Ubuntu 24.04, Gnome 46, nautilus removed and Nemo 6.0.2.

Strangely, a file drag and dropped from the desktop to a window of Google chrome appears in Google Chrome. Similarly, a URL link drag and drop from Google Chrome to a Nemo window appears on the desktop.

So definitely, the problem is on Nemo windows which is not "responding" to drag and drop.

### Steps to reproduce

Drag and drop a file from desktop to a nemo window.

### Expected behavior

File should be copied or moved the corresponding folder of the nemo window.

**Comment by rphair** [comment]
I can also confirm this: same versions as above.

This appears to relate to **window focus**: I'm using a manual arrangement of desktop icons, and when trying to drop the moved icon into the nemo window, it's ignored there (no focus) and simply dropped into that place on the nemo desktop background... i.e. it behaves as if the target nemo window isn't even there.

Workaround: cut from nemo desktop, paste to nemo folder (work the other way too).

**Comment by gregoiregentil** [comment]
Yes, copy paste is a workaround but after three months, I'm still trapped by the faulty drag-and-drop. And the icons are just moved as there was no nemo window...

**Comment by rphair** [comment]
Let's try posting on the Mint Forums, referencing this issue, and see if anyone knows a solution or a better workaround there. 🙏

**Comment by Jeremy7701** [comment]
Note:- Current stable version of nemo in Mint/LMDE is 6.2.8
nemo works for me on LMDE6.

**Comment by mtwebster** [comment]
Can't reproduce this in Cinnamon.

- Launch nemo
- Create a file on the desktop
- drag file into window
- file disappears from desktop, appears in window.

It may be some quirk with gnome-shell/mutter. Is this the wayland or x11 session? I can't make any guarantees under wayland at this time.

**Comment by rphair** [comment]
Guaranteed to happen on the default GNOME installation for Ubuntu 24.04:
```shell
$ echo $XDG_SESSION_TYPE
wayland
```

Yes indeed it could the Wayland "security" of the nemo desktop window not being allowed to interact with the nemo client window.

But if that's so, then it's strange that a file dragged from the nemo client window can be dropped in a precise location on the nemo desktop (though it's copied rather than moved... definitely a bug on Ubuntu).

I saw some nemo quirks like this on the short-lived Ubuntu 23.10 "Mantic" and they went away when disabling the Wayland session.  If there is a larger issue about Wayland support from Nemo then please @mtwebster let us know and I'll ask our user base to subscribe for updates there.

**Comment by mtwebster** [comment]
We have wayland 'catch-all' repo just for tracking wayland bugs:
https://github.com/linuxmint/wayland/issues

I'll move this issue there - any bug in x11 can go in that project's repo. Anything that occurs in wayland should go there instead.

As far as support in Nemo under wayland, when we implemented initial wayland support for Cinnamon, I had to add some special handling for nemo-desktop windows (which *must* use xwayland at this time). There are a number of commits related to this (https://github.com/linuxmint/muffin/commits/master/src/compositor/compositor.c has most of it).

mutter/gnome-shell won't have any of this (I think they render the desktop using gnome-shell itself now).
